src/Plugin/KeyType/ApigeeAuthKeyType.php

```diff
@@ -130,12 +135,15 @@ public function validateKeyValue(array $form, FormStateInterface $form_state, $k
    */
   public function getAuthenticationMethod(KeyInterface $key): Authentication {
     $values = $key->getKeyValues();
-
     if ($this->getInstanceType($key) === EdgeKeyTypeInterface::INSTANCE_TYPE_HYBRID) {
-      $account_key = $this->getAccountKey($key);
-      return new HybridAuthentication($account_key['client_email'], $account_key['private_key'], \Drupal::service('apigee_edge.authentication.oauth_token_storage'));
+      if ($this->useGcpDefaultServiceAccount($key)) {
+        return new GceServiceAccountAuthentication(\Drupal::service('apigee_edge.authentication.oauth_token_storage'));
+      }
+      else {
+        $account_key = $this->getAccountKey($key);
+        return new HybridAuthentication($account_key['client_email'], $account_key['private_key'], \Drupal::service('apigee_edge.authentication.oauth_token_storage'));
```

Comment:
Service should be Di-d.

Reply:
Idea: Create a HybridAuthenticationFactory for example to handle dependencies that should be constructor inject and those that vary per context, so they should be injected via method parameters.